hyperspy/io_plugins/emd.py

```diff
@@ -404,14 +421,581 @@ def log_info(self):
+            offsets['x'], original_units, spectrum_image_shape[1])
+        offset_y = self._convert_scale_units(
+            offsets['y'], original_units, spectrum_image_shape[0])
+
+        i = 0
+        axes = []
+        # add a supplementary axes when we import all frames individualy
+        if self.individual_frame:
+            frame_time = float(original_metadata['Scan']['FrameTime'])
+            axes.append({'index_in_array': i,
+                         'name': 'Time',
+                         'offset': 0,
+                         'scale': frame_time,
+                         'size': spectrum_image_shape[i],
+                         'units': 's'})
+            i += 1
+        axes.extend([{'index_in_array': i,
+                      'name': 'y',
+                      'offset': offset_y[0],
+                      'scale': scale_y[0],
+                      'size': spectrum_image_shape[i],
+                      'units': scale_y[1]},
+                     {'index_in_array': i + 1,
+                      'name': 'x',
+                      'offset': offset_x[0],
+                      'scale': scale_x[0],
+                      'size': spectrum_image_shape[i + 1],
+                      'units': scale_x[1]},
+                     {'index_in_array': i + 2,
+                      'name': 'X-ray energy',
+                      'offset': offset,
+                      'scale': dispersion,
+                      'size': spectrum_image_shape[i + 2],
+                      'units': 'keV'}])
+
+        md = self._get_metadata_dict()
+        md['Signal']['signal_type'] = 'EDS_TEM'
+
+        if self.individual_detector is False:
+            self.dictionaries.append({'data': streams.sum_spectrum_image,
+                                      'axes': axes,
+                                      'metadata': md,
+                                      'original_metadata': original_metadata,
+                                      'mapping': self._get_mapping()})
+        else:
+            for stream in streams.stream_list:
+                print('******************')
+                md['General']['title'] = 'EDS - {}'.format(stream.original_metadata[
+                    'BinaryResult']['Detector'])
+                print(md['General']['title'])
+                self.dictionaries.append({'data': stream.spectrum_image,
+                                          'axes': axes,
+                                          'metadata': md,
+                                          'original_metadata': stream.original_metadata,
+                                          'mapping': self._get_mapping()})
+
+    def _get_dispersion_offset(self, original_metadata):
+        for detectorname, detector in original_metadata['Detectors'].items():
+            _logger.debug('Detector: %s' % detector['DetectorName'])
+            if detector['DetectorName'] == 'SuperXG21' or detector['DetectorName'] == 'SuperXG11':
```

Comment:
You shouldn't assume that SuperXG21 is available as a detector name for 2 reasons:
1. In case of the SuperXG2 Lite detector (2 segment), these are called SuperXG2Lite3 and SuperXG2Lite4.
2. Segment 1 may be missing as a detector all together if it was (temporarily) disabled in the microscope configuration.

It's better to use ```BinaryResult.Detector``` to determine the key here. At least for G2, the name of the segment used to acquire this stream is in that field. So if you have independently acquired segments, each stream will have a unique segment name here.